### DISCUSSION TITLE
Explain in the docs best way to handle component API changes

### DISCUSSION THREAD

**Discussion by Oudwins** [huntabyte/shadcn-svelte#2034] [Help]
### Change Type

Addition

### Proposed Changes

Hey! 

I'm in the process of migrating shadcn-svelte to svelte 5. There are a lot of API changes on the components and it would be nice to have some examples on best ways to migrate those. Specially for when a prop was removed for example:

```svelte
            <DropdownMenu.Item
              class="cursor-pointer"
              href={$org + routes.team}
            >
              <UserCog class="mr-2 h-4 w-4" />
              <span>Manage Members</span>
            </DropdownMenu.Item>
```

That dropdown menu item is invalid because the new dropdown menu item no longer has the href prop, the asChild prop has also being removed. So to get similar behaviour to before I am having to make an a tag the child of the item and do some weird stuff around paddings. Which is fine but it would be nice to have a clear "this is the way we recommend migrating the components"

Again this is just an example, as I find more I am happy to add them here.

**Comment by Oudwins** [comment]
edit: found someone suggesting a way to do this:

https://github.com/huntabyte/bits-ui/issues/76#issuecomment-2477648425


it would be nice for this to be documented

**Comment by divmgl** [comment]
+1000 

The docs need to be updated to show _why_ `asChild` is no longer available. If it's available in `bits-ui`, then I expect it to be available in `shadcn-svelte`. shadcn/ui in React works the same way (with Radix under the hood).

Even just an example would be helpful! Thank you!

**Comment by jkschy** [comment]
+100

I am struggling to figure out a way to migrate all my components I have created that depend on the `asChild` and builders. These are now rendering as button in buttons which is not something I want. Is there a way to achieve the same previous functionality of using a "slotted" button, now a `Snippet`.
